### PR TITLE
Fix k Shortest Paths with weights, reactivate cluster tests

### PR DESCRIPTION
**Pull request by markuspf**
Need to use the closest neighbour, not the first one when intersection between the two neighbourhoods is found.

This only showed up in the cluster tests because the neighbours/edges appeared in a different order.

### CODE REVIEW

**Comment by markuspf** [comment]
Might be worth adapting the tests so that we tickle this as well, I'll keep a note in my Catch-test-TODO.

**Comment by mchacki** [comment]
Wait this also needs to modify the  test class to remove `-noncluster` from the file name.

**Comment by markuspf** [comment]
The code also is still buggy, working on a fix now.

**Comment by markuspf** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/4094/

File: arangod/Graph/KShortestPathsFinder.cpp [hunk 1]
```diff
@@ -155,9 +155,10 @@ bool KShortestPathsFinder::advanceFrontier(Ball& source, Ball const& target,
                                            std::unordered_set<Edge> const& forbiddenEdges,
                                            VertexRef& join) {
   VertexRef vr;
-  FoundVertex* v;
+  FoundVertex *v, *w;
 
   bool success = source._frontier.popMinimal(vr, v);
+  TRI_ASSERT(vr == v->_vertex);
```
Comment (mchacki) [inline arangod/Graph/KShortestPathsFinder.cpp]: please:
TRI_ASSERT(v != nullptr);
to avoid udef that we do not see. Before accessing v->_vertex

Comment (markuspf) [reply arangod/Graph/KShortestPathsFinder.cpp]: Done.

**Pull request merged**